ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionViewModelTest.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection
+
+import androidx.lifecycle.SavedStateHandle
+import com.google.android.ground.BaseHiltTest
+import com.google.common.truth.Truth.assertThat
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class DataCollectionViewModelTest : BaseHiltTest() {
+  @Inject lateinit var dataCollectionViewModelFactory: DataCollectionViewModel.Factory
+
+  @Test
+  fun savedState_containsCurrentPosition_returnsCorrectCurrentPosition() {
```

Comment:
Can you add tests for the null case and for the case when the task list changes?

Reply:
Hmm... do you have suggestions for how to emit a new value for the submission in the test to trigger the DataCollectionViewPager recreation?

Reply:
Ah nvm, that happens in the fragment, not the vm. We should eventually add instrumented tests to cover that case.